### PR TITLE
feat: added a new flow to let users manually configure transactions' gas limit on the System Contract DApp

### LINKED ISSUE #668: [System Contract DApp] Mark manual gasLimit input optional
### Problem

Currently, within the System Contract Dapp Playground, specific methods require users to manually input a gas amount in order to execute transactions. This manual input process can potentially introduce inefficiencies and inconvenience for users interacting with the platform. 

### Solution

Utilize ethers's estimateGas() method to automatically estimate gas for the transactions or use mirror node to query for estimated gas.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
This PR has implemented a new gas estimation mechanism for the System Contract DApp Playground. With this enhancement, users can now specify custom gas limits for transactions manually. In cases where no gas limits are provided, the transactions' calldata will be forwarded to a mirror-node for calculating the necessary gas limit. This improved flow is now accessible to all transactions undergoing consensus.

**Related issue(s)**:

Fixes #668 

**UI Updates**:
### Transactions will now have an extra textbox next to their execute buttons which receive custom gas limits.
image

###
image

###
image

###
image

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: system-contract-dapp-playground/__tests__/hedera/erc20-interactions/index.test.ts [hunk 1]
```diff
@@ -26,7 +26,12 @@ import {
   handleErc20TokenPermissions,
 } from '@/api/hedera/erc20-interactions';
 import { Contract } from 'ethers';
-import { MOCK_TX_HASH } from '../../utils/common/constants';
+import {
+  MOCK_TX_HASH,
+  MOCK_GAS_LIMIT,
+  MOCK_HEDERA_NETWORK,
+  MOCK_SINGER_ADDRESS,
```
Comment (Nana-EC) [inline system-contract-dapp-playground/__tests__/hedera/erc20-interactions/index.test.ts:33]:
```suggestion
  MOCK_SIGNER_ADDRESS,
```

**Review by Nana-EC — COMMENTED** [review]
LG
Minor edit but repeated many times

**Review by acuarica — APPROVED** [review]
lgtm, left some details comments

File: system-contract-dapp-playground/__tests__/hedera/erc20-interactions/index.test.ts (hunk 1 above)
Comment (quiet-node) [reply system-contract-dapp-playground/__tests__/hedera/erc20-interactions/index.test.ts:33]: Oh thanks for the nice catch!

File: system-contract-dapp-playground/src/api/hedera/hts-interactions/tokenTransfer-interactions/index.ts [hunk 2]
```diff
@@ -39,12 +45,25 @@ import { ISmartContractExecutionResult } from '@/types/contract-interactions/sha
...
   try {
+    if (gasLimit === 0) {
+      const estimateGasResult = await handleEstimateGas(
+        baseContract,
+        signerAddress,
+        network,
+        'cryptoTransferPublic',
+        [transferList, tokenTransferList]
+      );
+      if (!estimateGasResult.gasLimit || estimateGasResult.err) return { err: estimateGasResult.err };
```
[the remark this answers is not in the document]
Comment (quiet-node) [reply system-contract-dapp-playground/src/api/hedera/hts-interactions/tokenTransfer-interactions/index.ts:64]: That's a great suggestion Luis, I appreciate it! You're right this logic recurs in most of the API calls. I've devised another plan to address this by crafting a more robust error handler, which I'll integrate into another PR. 😄

**Review by acuarica — APPROVED** [review]
lg

**Pull request merged**